### PR TITLE
#1462 extracted MapGrid's HasGravity property to separate Content com…

### PR DESCRIPTION
**Pull request by GalacticChimp**
…ponent

## About the PR 

Fixes https://github.com/space-wizards/RobustToolbox/issues/1462
Requires https://github.com/space-wizards/RobustToolbox/pull/1835
Requires https://github.com/space-wizards/RobustToolbox/pull/1838

### CODE REVIEW

File: Content.Server/Gravity/EntitySystems/GravitySystem.cs [hunk 1]
```diff
@@ -48,16 +48,17 @@ public override void Update(float frameTime)
 
             foreach (var grid in _mapManager.GetAllGrids())
             {
-                if (grid.HasGravity && !gridsWithGravity.Contains(grid.Index))
+                var gridEntity = EntityManager.GetEntity(grid.GridEntityId);
```
Comment (GalacticChimp) [inline Content.Server/Gravity/EntitySystems/GravitySystem.cs:51]: I'm not sure if this is the correct pattern for checking components existence on Entities

File: Content.Shared/Gravity/GravityComponent.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+using Robust.Shared.GameObjects;
+
+namespace Content.Shared.Gravity
+{
+    [RegisterComponent]
+    public class GravityComponent : Component
```
Comment (GalacticChimp) [inline Content.Shared/Gravity/GravityComponent.cs:6]: an empty, marker component should be enough?

Comment (mirrorcult) [reply Content.Shared/Gravity/GravityComponent.cs:6]: usually marker components are a sign of bad but this seems fine

Comment (mirrorcult) [reply Content.Shared/Gravity/GravityComponent.cs:6]: an idea Vera had was having a 'SpaceStationComponent' which you attach to a grid to signify that it's a space station; which would hold stuff like the manifest, station name, and i guess that it has gravity?

File: Content.Server/Gravity/EntitySystems/GravitySystem.cs [hunk 3]
```diff
@@ -48,16 +48,17 @@ public override void Update(float frameTime)
...
+                if (gridEntity.HasComponent<GravityComponent>() && !gridsWithGravity.Contains(grid.Index))
                 {
                     DisableGravity(grid);
                 }
-                else if (!grid.HasGravity && gridsWithGravity.Contains(grid.Index))
+                else if (!gridEntity.HasComponent<GravityComponent>() && gridsWithGravity.Contains(grid.Index))
```
Comment (metalgearsloth) [inline Content.Server/Gravity/EntitySystems/GravitySystem.cs:56]: Could avoid doing the HasComponent check twice.

File: Content.Shared/Gravity/GravityComponent.cs (hunk 2 above)
Comment (metalgearsloth) [reply Content.Shared/Gravity/GravityComponent.cs:6]: This also needs to be used for shuttles so having the additional data for them seems silly given it's superfluous.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Review by metalgearsloth — APPROVED** [review]

**Comment by metalgearsloth** [comment]
Okay I actually played it in-game and this is currently unmergeable due to rubberbanding so I'll fix it.

**Comment by metalgearsloth** [comment]
I made some changes and PRd them to your fork but needs https://github.com/space-wizards/RobustToolbox/pull/1838

Also sussing out why AddComponent isn't syncing from server to client.

**Pull request merged**